### PR TITLE
New label code - fixes #783, #784, #797, #799

### LINKED ISSUE #783: [discussion] Simplify label/prefix handling with a proposed new LabelList model
From IRC by @hallvors 

> I have a proposal, sort of: https://gist.github.com/hallvors/9cdf228aeb72598b78eb for a `LabelList` model initialized by either URL or array. Just `list.get('labels')` to get an array of objects fit to present in the page (`list[i].name` and `list[i].color` ready for your templates). just `list.save()` to send stuff to backend - prefixing should just happen. But it's not well tested yet :-p. Didn't really use the Vanilla.js framework ;). (when adding labels - when new things are checked, for example - it accepts `list.set('labels', listOfObjects)`). If you like that, I'll rewrite all code touching label stuff to use `LabelList`.

> @miketaylr said: "i'll leave some comments in the [gist](https://gist.github.com/hallvors/9cdf228aeb72598b78eb)"

### PR DESCRIPTION
**Pull request by hallvors**
r?@miketaylr

### CODE REVIEW

**Comment by miketaylr** [comment]
https://github.com/webcompat/webcompat.com/blob/master/tests/test_api_urls.py#L67 needs to be updated.

File: webcompat/static/js/lib/labels.js [hunk 1]
```diff
@@ -4,31 +4,119 @@
...
+*
+* new issues.LabelList({labels: [{name:'status-worksforme', url:'...',
+*    color:'cccccc'}]});
+*
+* Or a URL to a JSON file describing the labels:
+*
+* new issues.LabelList({url:'/path/to/labels.json'});
+*/
+
+issues.LabelList = Backbone.Model.extend({
```
Comment (miketaylr) [inline webcompat/static/js/lib/labels.js]: Now that the label model is more than 5 lines (or whatever), can we split this out and add to `/lib/models/labels.js` or `/lib/models/label-list.js`?

**Comment by miketaylr** [comment]
Looks good to me, just a few things to fix:

- [ ] move label model to own file
- [ ] update failing test
- [ ] teach @api.route('/issues/labels') how to do a proxy request for non-auth users

**Comment by hallvors** [comment]
(On the /api/labels requests, we're probably moving towards having a set of labels "hard-coded" in config, if we can land that soon we won't be using Twitter API here at all I think. But I'll figure out how to make it a "proxy call" meanwhile.)

**Comment by miketaylr** [comment]
Take a look at the `/rate_limit` endpoint, that's a pretty simple example of how to do it.

**Comment by hallvors** [comment]
Ready to merge!

**Comment by hallvors** [comment]
I'm waiting to see if GitHub is smart enough to auto-close all three issues mentioned in summary when @miketaylr merges this :)

**Comment by miketaylr** [comment]
Awesome, thanks @hallvors.

**Pull request merged**